File: Forge/src/main/java/hantonik/atomic/core/block/entity/IAtomicPacketHandler.java
package hantonik.atomic.core.block.entity;

import net.minecraft.network.FriendlyByteBuf;

public interface IAtomicPacketHandler {
    default FriendlyByteBuf getMenuPacket(FriendlyByteBuf buffer) {
        return buffer;
    }

    default void handleMenuPacket(FriendlyByteBuf buffer) {}

    default FriendlyByteBuf getStatePacket(FriendlyByteBuf buffer) {
        return buffer;
    }

    default void handleStatePacket(FriendlyByteBuf buffer) {}
}

File: Forge/src/main/kotlin/hantonik/atomic/core/block/entity/IAtomicPacketHandler.kt
package hantonik.atomic.core.block.entity

import net.minecraft.network.FriendlyByteBuf

interface IAtomicPacketHandler {
    fun getMenuPacket(buffer: FriendlyByteBuf) = buffer

    fun getStatePacket(buffer: FriendlyByteBuf) = buffer

    fun handleMenuPacket(buffer: FriendlyByteBuf) {}

    fun handleStatePacket(buffer: FriendlyByteBuf) {}
}